src/fallbacks.jl

```diff
@@ -44,25 +44,32 @@ haslength(L) = L isa Union{Base.HasShape, Base.HasLength}
     
     Custom column types can override with an appropriate "scalar" element type that should dispatch to their column allocator.
 """
-allocatecolumn(T, len) = Vector{T}(undef, len)
+function allocatecolumn(::Type{T}, len; unwrap = t -> false) where {T}
```

Comment:
mhm, actually if this is supposed to be overloaded (say to get a `DataValueArray` or things like that), I should add an extra step in between and not touch this one. So `allocatecolumns` should call some `_allocatecolumn` that decides whether calling `allocatecolumn` or to call back `allocatecolumns`.